### PR TITLE
Refactor default dropdown css

### LINKED ISSUE #265: Update default CSS for the dropdown component
#### Used in several places
billing cycle:
image

cluster list:
image

cluster list console:
image

what a default dropdown looks like (you can see size disrepency compared to the cluster list dropdown, look at storybook for more)
image

For each of these we duplicate almost the same CSS. We need to update the default dropdown css instead so there is fewer overrides when extending from default.  The new default item size needs to be the same as in the cluster dropdown.

The `default` css for dropdowns is the one defined in: https://github.com/gravitational/webapps/blob/master/packages/shared/components/Select/Select.tsx. We want this default css to be the same as the css for our cluster dropdown in terms of sizing, spacing, fonts, hover color (we are keeping the white bg though). Add pointer cursor on hovers too.

Billing cycle dropdown should be converted to the default (white background) as well. The cluster list will remain the same look.

### PR DESCRIPTION
**Pull request by amha19**
**Description**

This PR refactors the default select component styling.
Creates a new styling component to override the default select component for darker themes. 

The purpose is to reduce code duplication, have less code overriding the default styling and consistent styling across components that use the default drop-down.

This PR resolves #265 

### Screenshot
image

### CODE REVIEW

File: packages/shared/components/Select/Select.story.tsx
Comment (kimlisa) [inline packages/shared/components/Select/Select.story.tsx:86]: this should be a dark default, otherwise every time we use `DarkStyledSelect`, we have to repeat this style

File: packages/shared/components/Select/Select.story.tsx
Comment (kimlisa) [inline packages/shared/components/Select/Select.story.tsx:1]: since we have this file to showcase light vs dark, lets create a normal searchable selector too (not multis)

**Comment by kimlisa** [comment]
Couple of things:

- we wanted to retain the coloring of what it was before. See the current and before image for the cluster selector. We should just make it `theme.colors.text.primary`. 
- The placeholder text is still grey and I didn't see this addressed in our dark styled component, i think we agreed on `theme.colors.text.secondary`

Current:
image

This is before:
image

File: packages/shared/components/Select/Select.story.tsx
Comment (kimlisa) [inline packages/shared/components/Select/Select.story.tsx:103]: I'd put a note here saying why we increased the height

File: packages/shared/components/Select/Select.story.tsx
Comment (kimlisa) [inline packages/shared/components/Select/Select.story.tsx:23]: Since its just one file for now, rename to `Shared`, and rename `SelectTypes` as `Selects` so there is no nesting

File: packages/teleport/src/Console/DocumentNodes/ClusterSelector/ClusterSelector.tsx
Comment (kimlisa) [inline packages/teleport/src/Console/DocumentNodes/ClusterSelector/ClusterSelector.tsx:112]: lets add back the `color: white`, i did a side by side test, and the white looks better on the very dark bg

File: packages/shared/components/Select/DarkStyledSelect.tsx
Comment (kimlisa) [inline packages/shared/components/Select/DarkStyledSelect.tsx:64]:
```suggestion
  .react-select__control--menu-is-open {
```
I think this is better, it only highlights when its opened, and removes focus afterwards

**Review by kimlisa — COMMENTED** [review]
could we also update this issue with a image of the selects

File: packages/shared/components/Select/DarkStyledSelect.tsx
Comment (amha19) [reply packages/shared/components/Select/DarkStyledSelect.tsx]: this will do the trick

**Review by kimlisa — APPROVED** [review]
please wait for @alex-kovoy approval

**Review by kimlisa — CHANGES_REQUESTED** [review]
I don't think we handled disabled state. I would put another side by side of disabled selects. Example below is a disabled state but doesn't appear disabled.

image

**Comment by amha19** [comment]
Disabled state for default and dark style selects. 

disabled-default-dark

**Review by kimlisa — APPROVED** [review]

**Review by alex-kovoy — APPROVED** [review]

**Pull request merged**
